shared/api/instance.go

```diff
@@ -49,6 +49,13 @@ type InstancesPost struct {
 	// Type (container or virtual-machine)
 	// Example: container
 	Type InstanceType `json:"type" yaml:"type"`
+
+	// Whether we want to rebuild the instance
+	Rebuild bool `json:"rebuild" yaml:"rebuild"`
+
+	// If `Rebuild` is true, `NewSource` holds the new source to use.
+	// If empty, the instance will be rebuilt using its current instance source.
+	NewSource InstanceSource `json:"new_source" yaml:"new_source"`
```

Comment:
If we are using the existing struct, and not a new endpoint (as the spec suggests), then we don't need this field and can reuse the existing `Source` field when `Rebuild` is true.